### PR TITLE
Add GVRRenderer, C++ Renderer class no longer a singleton

### PR DESCRIPTION
**Pull request by NolaDonato**
Renderer now derives from HybridObject and will be garbage collected with the render bundle. Added IRenderBundle.getRenderer().

This may fix the crash on pause / resume.

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

### CODE REVIEW

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRShader.java [hunk 1]
```diff
@@ -51,7 +51,7 @@
  */
 public class GVRShader
 {
-    protected boolean mWriteShadersToDisk = false;
+    protected boolean mWriteShadersToDisk = true;
```
Comment (apjagdale) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRShader.java]: Needs to be false

**Comment by liaxim** [comment]
gvr-blurfilter and gvr-rendertotexture crash:
```
10-17 12:28:54.814 24370 24446 E AndroidRuntime: FATAL EXCEPTION: GLThread 444
10-17 12:28:54.814 24370 24446 E AndroidRuntime: Process: org.gearvrf.rendertotexture, PID: 24370
10-17 12:28:54.814 24370 24446 E AndroidRuntime: java.lang.UnsatisfiedLinkError: No implementation found for void org.gearvrf.NativeRenderer.cullAndRender(long, long, long, long, long, long) (tried Java_org_gearvrf_NativeRenderer_cullAndRender and Java_org_gearvrf_NativeRenderer_cullAndRender__JJJJJJ)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.NativeRenderer.cullAndRender(Native Method)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRRenderer.cullAndRender(GVRRenderer.java:41)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRViewManager.cullAndRender(GVRViewManager.java:521)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRRenderTarget.onDrawFrame(GVRRenderTarget.java:142)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRViewManager.doMemoryManagementAndPerFrameCallbacks(GVRViewManager.java:261)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRViewManager.access$600(GVRViewManager.java:46)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRViewManager$2.beforeDrawEyes(GVRViewManager.java:381)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.GVRViewManager.beforeDrawEyes(GVRViewManager.java:493)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.OvrViewManager.beforeDrawEyes(OvrViewManager.java:205)
10-17 12:28:54.814 24370 24446 E AndroidRuntime: 	at org.gearvrf.OvrViewManager.onDrawFrame(OvrViewManager.java:294)
```

**Comment by liaxim** [comment]
Should we close?

**Comment by NolaDonato** [comment]
Yes. GVRRenderer might be a good idea but the overall thing needs some rethinking.

**Pull request closed without merging**